--- tcptrack-1.4.3.orig/src/TextUI.cc	2015-04-23 11:04:40.000000000 +0300
+++ tcptrack-1.4.3.new/src/TextUI.cc	2021-12-16 17:23:57.548316417 +0200
@@ -309,11 +309,11 @@
 
 		move(row,58);
 		if( ic->getIdleSeconds() < 60 )
-			printw("%ds",ic->getIdleSeconds());
+			printw("%ds",(int)(ic->getIdleSeconds()));
 		else if( ic->getIdleSeconds() > 59 ) 
-			printw("%dm",ic->getIdleSeconds()/60);
+			printw("%dm",(int)(ic->getIdleSeconds()/60));
 		else if( ic->getIdleSeconds() > 3559 )
-			printw("%dh",ic->getIdleSeconds()/3600);
+			printw("%ldh",(long)(ic->getIdleSeconds()/3600));
 
 		move(row,63);
 		if( ic->activityToggle() )
